src/index.rs

```diff
@@ -180,20 +200,25 @@ impl Index {
     }
 
     pub fn find_type(&self, type_name: &str) -> Option<&DataType> {
-        let data_type = self.types.get(type_name);
-        data_type.map(|it| {
-            if let DataTypeInformation::Alias { referenced_type, .. } = it.get_type_information() {
-                return self.find_type(referenced_type.as_str());
-            }
-            Some(it)
-        }).flatten()
+        self.types.get(type_name)
     }
 
     
     pub fn get_type(&self, type_name: &str) -> Result<&DataType, CompileError> {
         self.find_type(type_name).ok_or_else(|| CompileError::unknown_type(type_name, 0..0))
     }
 
+
+    /// Retrieves the "Primitive" type behind this datatype
+    /// A primitive type will be any end type i.e. Structs, Integers, Floats, String and Array
+    pub fn find_primitive_type<'ret>(&'ret self, data_type : &'ret DataTypeInformation) -> Option<&'ret DataTypeInformation> {
```

Comment:
I think primitive rather refers to the most basic types like byte, int, bool

what do you think about: find_effective_type...?

Reply:
I was thinking of [first-class](https://llvm.org/docs/LangRef.html#t-firstclass) types from LLVM which also include structs and arrays, for some reason I rememberd them as "primitive".
I could change it to effective

Reply:
I understand, when I google "primitive Type" it looks like google agrees with me.
the wiki page to Primitive Data Types says composite datatypes are made of primitive data types or other composites